tensorboard/plugins/debugger_v2/tf_debugger_v2_plugin/store/debugger_reducers.ts

```diff
@@ -564,6 +567,82 @@ const reducer = createReducer(
       return newState;
     }
   ),
+  on(
+    actions.graphExecutionDataRequested,
+    (state: DebuggerState, {pageIndex}): DebuggerState => {
+      if (state.activeRunId === null) {
+        return state;
+      }
+      const graphExecutionDataLoadingPages = state.graphExecutions.graphExecutionDataLoadingPages.slice();
+      if (graphExecutionDataLoadingPages.indexOf(pageIndex) === -1) {
+        graphExecutionDataLoadingPages.push(pageIndex);
+      }
+      return {
+        ...state,
+        graphExecutions: {
+          ...state.graphExecutions,
+          graphExecutionDataLoadingPages,
+        },
+      };
+    }
+  ),
+  on(
+    actions.graphExecutionDataLoaded,
+    (state: DebuggerState, data: GraphExecutionDataResponse): DebuggerState => {
+      if (state.activeRunId === null) {
+        return state;
+      }
+      const {pageSize} = state.graphExecutions;
+      const graphExecutionDataLoadingPages = state.graphExecutions.graphExecutionDataLoadingPages.slice();
+      const graphExecutionDataPageLoadedSizes = {
+        ...state.graphExecutions.graphExecutionDataPageLoadedSizes,
+      };
+      const graphExecutionData = {...state.graphExecutions.graphExecutionData};
+      for (let i = data.begin; i < data.end; ++i) {
+        const pageIndex = Math.floor(i / pageSize);
+        if (graphExecutionDataLoadingPages.indexOf(pageIndex) !== -1) {
+          graphExecutionDataLoadingPages.splice(
+            graphExecutionDataLoadingPages.indexOf(pageIndex),
+            1
+          );
+        }
```

Comment:
`graphExecutionDataLoadingPages` maintains the pages that are currently being loaded. An object could serve the same purpose, perhaps a little more performantly if the number of pages is large. But the downside is that the key type will be a little arbitrary (perhaps just `null`?) We don't want to keep track of all the pages, because the number of pages can be large (consider a debugger run with 500k graph executions, which leads to 500k / 200 = 2500 keys in the object if we did that). In practice, the performance shouldn't matter either, because the amount of loading pages should usually be small (the size of the array grows with user scrolling the list, at a debounced rate).